### ISSUE TITLE
last_over_time doesn't return the actual timestamp of the last sample

### ISSUE THREAD

**Issue by dandv** [VictoriaMetrics/VictoriaMetrics#5709]
### Describe the bug

I want to get the actual timestamp of the last sample in a series. I thought an instant query would return that, but [nope](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5708). Surprisingly, `last_over_time` with a duration didn't return the actual timestamp either. The [documentation](https://docs.victoriametrics.com/metricsql/#last_over_time) says it

> returns the last raw sample value on the given lookbehind window `d`

but it actually returns the `time` parameter passed to the [instant query](https://docs.victoriametrics.com/keyconcepts/#instant-query), so this looks like a bug.

### To Reproduce

1. curl http://localhost:8428/api/v1/import/prometheus -d 'series1 123'
2. curl http://localhost:8428/api/v1/query -d 'query=last_over_time(series1[1h])' | jq  # notice the current timestamp in seconds
3. curl http://localhost:8428/api/v1/query -d 'query=series1[1h]' | jq  # the actual timestamp is returned

### Version

victoria-metrics-20231212-224437-tags-v1.96.0-0-g304fe0565

### Screenshots

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/33569/b7298fe3-c84e-4d22-ab6d-75951f859460)

### Used command-line flags

-search.latencyOffset=1s

**Comment by Haleygo** [comment]
Hello!
>Surprisingly, last_over_time with a duration didn't return the actual timestamp either.

No, it's expected.
When you perform [instant query](https://docs.victoriametrics.com/keyconcepts/#instant-query) without param `time`, the `time` is set to `now()` and the timestamp in result samples must be `now()`. And as you metioned, the description of `last_over_time` is `returns the last raw sample value` instead of `returns the last raw sample timestamp`. 
If  you want to get `last raw sample timestamp`, try [timestamp(series_selector[d])](https://docs.victoriametrics.com/metricsql/#timestamp).

**Comment by hagen1778** [comment]
Related to https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5708#issuecomment-1914471003
and b5978ed8f9e2d46e3f1358152610a760f4359102

**Comment by dandv** [comment]
Thank you for clarifying. What is the recommended method to get the actual last raw sample/point from a series, including its timestamp and value?

**Comment by hagen1778** [comment]
> Thank you for clarifying. What is the recommended method to get the actual last raw sample/point from a series, including its timestamp and value?

The last value can be queried via [last_over_time](https://docs.victoriametrics.com/metricsql/#last_over_time).
The last timestamp can be queried via [tlast_over_time](https://docs.victoriametrics.com/metricsql/#tlast_over_time)
Instant query also allows specifying vector next to time series selector to get raw values on the interval `foo[interval]`.

**Comment by dandv** [comment]
Specifying the [interval] after the series selector requires knowing how far behind to look. If `interval` is too large, a lot of samples will be needlessly loaded into memory in order to only access the last one. If it's too small, no samples will be returned.

The export endpoint requires knowing the `start`, and doesn't provide a way to limit the number of samples returned.

Looks like the last sample just cannot be obtained efficiently in one query?

**Comment by hagen1778** [comment]
> Looks like the last sample just cannot be obtained efficiently in one query?

Right, it can't be obtained without specifying the search interval.

Getting the last value is usually not needed in infrastructure monitoring or APM. But it could be useful for IoT usecases. There were a feature request to introduce a new endpoint to get the last value here https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2699

Could you please elaborate more on your usecase?

**Comment by dandv** [comment]
One common use case is for a display agent to show the last value gathered by a collection agent. Examples:
* the last heart rate measurement for a wearable
* the last temperature/pressure recorded
* the last price of a financial asset

**Comment by Haleygo** [comment]
Close this ticket as answered in https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5709#issuecomment-1914970889 and https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5709#issuecomment-1916307699.
As for the asked feature: get the last sample efficiently in one query, see similar request in https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2699.
